src/aleph/vm/controllers/qemu/snapshot_manager.py

```diff
@@ -0,0 +1,121 @@
+import asyncio
+import logging
+import threading
+from time import sleep
+
+from aleph_message.models import ItemHash
+from schedule import Job, Scheduler
+
+from aleph.vm.conf import settings
+from aleph.vm.controllers.firecracker.snapshots import CompressedDiskVolumeSnapshot
+
+logger = logging.getLogger(__name__)
+
+
+def wrap_async_snapshot(vm):
+    asyncio.run(do_vm_snapshot(vm))
+
+
+def run_threaded_snapshot(vm):
+    job_thread = threading.Thread(target=wrap_async_snapshot, args=(vm,))
+    job_thread.start()
+
+
+async def do_vm_snapshot(vm) -> CompressedDiskVolumeSnapshot:
+    try:
+        logger.debug(f"Starting new snapshot for QEMU VM {vm.vm_hash}")
+        assert vm, "VM execution not set"
+
+        snapshot = await vm.create_snapshot()
+        logger.debug(f"New snapshot for QEMU VM {vm.vm_hash} created successfully")
+        return snapshot
+    except ValueError as error:
+        msg = "Failed to create QEMU VM snapshot"
+        raise ValueError(msg) from error
+
+
+def infinite_run_scheduler_jobs(scheduler: Scheduler) -> None:
+    while True:
+        scheduler.run_pending()
+        sleep(1)
+
+
+class QemuSnapshotExecution:
+    vm_hash: ItemHash
+    execution: any  # AlephQemuInstance
```

Comment:
```suggestion
    execution: Any  # AlephQemuInstance
```

This should be `Any` from typing, I'm surprised this worked